src/plugin/parse-html/src/java/org/apache/nutch/parse/html/DOMContentUtils.java

```diff
@@ -116,36 +114,50 @@ public void setConf(Configuration conf) {
    * 
    * @return true if nested anchors were found
    */
-  public boolean getText(StringBuffer sb, Node node,
-      boolean abortOnNestedAnchors) {
-    if (getTextHelper(sb, node, abortOnNestedAnchors, 0)) {
+  private boolean getText(StringBuffer sb, Node node,
+                          boolean abortOnNestedAnchors, Set<String> excludedElementNames) {
+    if (getTextHelper(sb, node, abortOnNestedAnchors, 0, excludedElementNames)) {
       return true;
     }
     return false;
   }
 
   /**
    * This is a convinience method, equivalent to
-   * {@link #getText(StringBuffer,Node,boolean) getText(sb, node, false)}.
+   * {@link #getText(StringBuffer,Node,boolean, Set) getText(sb, node, false, excludedElementNames)}.
    * 
    */
-  public void getText(StringBuffer sb, Node node) {
-    getText(sb, node, false);
+  public void getText(StringBuffer sb, Node node, Set<String> excludedElementNames) {
+    getText(sb, node, false, excludedElementNames);
   }
 
   // returns true if abortOnNestedAnchors is true and we find nested
   // anchors
   private boolean getTextHelper(StringBuffer sb, Node node,
-      boolean abortOnNestedAnchors, int anchorDepth) {
+                                boolean abortOnNestedAnchors, int anchorDepth, Set<String> excludedElementNames) {
     boolean abort = false;
     NodeWalker walker = new NodeWalker(node);
+    Set<String> lcExcludedElementNames = new HashSet<>();
```

Comment:
We should avoid duplicating the exclusion set. This method is executed many times. We could use a `TreeSet` when it is invoked, delegating the comparison to the `Set` implementation.